web/src/components/Sidebar.tsx

```diff
@@ -1,25 +1,32 @@
+import { isUndefined } from "lodash-es";
+import { useEffect } from "react";
 import { Link } from "react-router-dom";
 import { useTranslation } from "react-i18next";
 import { userService } from "../services";
+import { useAppSelector } from "../store";
 import showDailyReviewDialog from "./DailyReviewDialog";
 import showSettingDialog from "./SettingDialog";
 import UserBanner from "./UserBanner";
 import UsageHeatMap from "./UsageHeatMap";
 import ShortcutList from "./ShortcutList";
 import TagList from "./TagList";
-import { closeSidebar } from "../helpers/utils";
 import "../less/siderbar.less";
 
 const Sidebar = () => {
   const { t } = useTranslation();
+  const location = useAppSelector((state) => state.location);
+
+  useEffect(() => {
+    toggleSiderbar(false);
+  }, [location.query]);
```

Comment:
When the `location.query` changed, usually caused by a tab/shortcut click, then we should hide the sidebar. @h2o2o